src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -854,6 +856,80 @@ ov::pass::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id, const bool s
     this->register_matcher(m, callback);
 }
 
+ov::pass::RoPEFusionChatGLMHF::RoPEFusionChatGLMHF() {
+    MATCHER_SCOPE(RoPEFusionChatGLMHF);
+
+    auto qk_linear = makePattern("[?,?,?]");
+    auto cos = makePattern("[?,?,?,?]");
+    auto sin = makePattern("[?,?,?,?]");
+
+    auto ndims = ov::gen_pattern::Symbol("ndims");
+    auto head_cnt = ov::gen_pattern::Symbol("head_cnt");
+    auto head_size = ov::gen_pattern::Symbol("head_size");
+
+    auto A = ov::gen_pattern::Symbol("A");
+    auto B = ov::gen_pattern::Symbol("B");
+    auto C = ov::gen_pattern::Symbol("C");
+
+    auto transpose = makePattern<opset1::Reshape>({qk_linear, {-1, head_cnt, 1, head_size}}, {{"special_zero", false}});
```

Comment:
too minor :) but why its transpose for Reshape op?

Reply:
The name of the pattern is created based on original node's name. However, It has been updated for readability